packages/flutter/lib/src/rendering/statistics_box.dart

```diff
@@ -34,27 +63,35 @@ class RenderStatisticsBox extends RenderBox {
   bool get sizedByParent => true;
 
   double getMinIntrinsicWidth(BoxConstraints constraints) {
-    assert(constraints.isNormalized);
-    return constraints.minWidth;
+    return constraints.constrainWidth(0.0);
   }
 
   double getMaxIntrinsicWidth(BoxConstraints constraints) {
-    assert(constraints.isNormalized);
-    return constraints.maxWidth;
+    return constraints.constrainWidth(0.0);
+  }
+
+  double get intrinsicHeight {
+    const kGraphHeight = 80.0; // must match value in statistics_layer.cc
+    double result = 0.0;
+    if ((optionsMask | (1 << StatisticsOption.displayRasterizerStatistics.index) > 0) ||
+        (optionsMask | (1 << StatisticsOption.visualizeRasterizerStatistics.index) > 0))
+      result += kGraphHeight;
+    if ((optionsMask | (1 << StatisticsOption.displayEngineStatistics.index) > 0) ||
+        (optionsMask | (1 << StatisticsOption.visualizeEngineStatistics.index) > 0))
+      result += kGraphHeight;
+    return result;
   }
 
   double getMinIntrinsicHeight(BoxConstraints constraints) {
-    assert(constraints.isNormalized);
-    return constraints.minHeight;
+    return constraints.constrainHeight(intrinsicHeight);
   }
 
   double getMaxIntrinsicHeight(BoxConstraints constraints) {
-    assert(constraints.isNormalized);
-    return constraints.maxHeight;
+    return constraints.constrainHeight(intrinsicHeight);
   }
 
   void performResize() {
-    size = constraints.biggest;
+    size = constraints.constrain(new Size(double.INFINITY, intrinsicHeight));
   }
 
   void paint(PaintingContext context, Offset offset) {
```

Comment:
Currently this class lies because it creates a layer but doesn't have the compositing bits set to say that it creates a layer.  Should we set hasLayer to true?  The risk is that turning on the performance overlay can change the compositing strategy for the app...